packages/anoncreds/src/anoncreds-rs/AnonCredsRsHolderService.ts

```diff
@@ -336,26 +505,69 @@ export class AnonCredsRsHolderService implements AnonCredsHolderService {
         methodName: options.methodName,
       })
 
-    return credentialRecords.map((credentialRecord) => ({
-      attributes: Object.fromEntries(
-        Object.entries(credentialRecord.credential.values).map(([key, value]) => [key, value.raw])
-      ),
-      credentialDefinitionId: credentialRecord.credential.cred_def_id,
-      credentialId: credentialRecord.credentialId,
-      schemaId: credentialRecord.credential.schema_id,
-      credentialRevocationId: credentialRecord.credentialRevocationId ?? null,
-      revocationRegistryId: credentialRecord.credential.rev_reg_id ?? null,
-      methodName: credentialRecord.methodName,
-    }))
+    return credentialRecords.map((credentialRecord) => getAnoncredsCredentialInfoFromRecord(credentialRecord))
   }
 
-  public async deleteCredential(agentContext: AgentContext, credentialId: string): Promise<void> {
-    const credentialRepository = agentContext.dependencyManager.resolve(AnonCredsCredentialRepository)
-    const credentialRecord = await credentialRepository.getByCredentialId(agentContext, credentialId)
-    await credentialRepository.delete(agentContext, credentialRecord)
+  public async getCredentials(
+    agentContext: AgentContext,
+    options: GetCredentialsOptions
+  ): Promise<AnonCredsCredentialInfo[]> {
+    const credentialRecords = await agentContext.dependencyManager
+      .resolve(W3cCredentialRepository)
+      .findByQuery(agentContext, {
+        anonCredsCredentialDefinitionId:
+          !options.credentialDefinitionId || isUnqualifiedCredentialDefinitionId(options.credentialDefinitionId)
+            ? undefined
+            : options.credentialDefinitionId,
+        anonCredsSchemaId: !options.schemaId || isUnqualifiedSchemaId(options.schemaId) ? undefined : options.schemaId,
+        anonCredsIssuerId: !options.issuerId || isUnqualifiedIndyDid(options.issuerId) ? undefined : options.issuerId,
+        anonCredsSchemaName: options.schemaName,
+        anonCredsSchemaVersion: options.schemaVersion,
+        anonCredsSchemaIssuerId:
+          !options.schemaIssuerId || isUnqualifiedIndyDid(options.schemaIssuerId) ? undefined : options.schemaIssuerId,
+
+        anonCredsMethodName: options.methodName,
+        anonCredsUnqualifiedSchemaId:
+          options.schemaId && isUnqualifiedSchemaId(options.schemaId) ? options.schemaId : undefined,
+        anonCredsUnqualifiedIssuerId:
+          options.issuerId && isUnqualifiedIndyDid(options.issuerId) ? options.issuerId : undefined,
+        anonCredsUnqualifiedSchemaIssuerId:
+          options.schemaIssuerId && isUnqualifiedIndyDid(options.schemaIssuerId) ? options.schemaIssuerId : undefined,
+        anonCredsUnqualifiedCredentialDefinitionId:
+          options.credentialDefinitionId && isUnqualifiedCredentialDefinitionId(options.credentialDefinitionId)
+            ? options.credentialDefinitionId
+            : undefined,
+      })
```

Comment:
How does it deal with false being passed? I think that can result in issues?

Maybe we can make this a bit more of a builder flow? Even if it works with Askar, we're basically filtering for tags having value `false`.

Reply:
what do you mean with false being passed?

Reply:
Nevermind. Didn't understand the code correctly